specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,276 @@
+<!-- USAGE
+  * Fill in each of the sections (like Background) below
+  * Wrap code with `single line of code` or ```code block```
+  * Before submitting, delete all <!-- TEMPLATE marked comments in this file,
+    and the following quote banner:
+-->
+> See comments in Markdown for how to use this spec template
+
+<!-- TEMPLATE
+  The purpose of this spec is to describe a new WebView2 feature and its APIs.
+
+  There are two audiences for the spec. The first are people
+  that want to evaluate and give feedback on the API, as part of
+  the submission process. When it's complete
+  it will be incorporated into the public documentation at
+  docs.microsoft.com (https://docs.microsoft.com/en-us/microsoft-edge/webview2/).
+  Hopefully we'll be able to copy it mostly verbatim.
+  So the second audience is everyone that reads there to learn how
+  and why to use this API.
+-->
+
+
+# Background
+By default HTTP basic authentication and NTLM authentication requests inside WebView2 show the authentication UI, which is a dialog prompt in which the user can type in user name and password credentials just like in the Edge browser. We have been requested by WebView2 app developers to provide finer granularity for managing HTTP Basic/NTLM authentications inside WebView2, including the ability to hide the login UI and provide credentials.
+
+
+# Description
+We propose a new event for WebView2, CoreWebView2BasicAuthenticationRequested that will allow developers to listen on and override the HTTP Basic authentication requests in WebView2. When there is a HTTP Basic/NTLM authentication request in WebView2, the developer will have a choice to:
```

Comment:
The Requested event pattern doesn't seem to quite fit, but I don't have an alternative to propose.

I don't remember seeing a Requested event pattern being used when there's pending default behavior. So it looks unusual that you can not handle a Requested event, and yet everything works.

And the Cancel pattern almost always goes with an Ing ("about to happen") event. This case is more like the Handled pattern, like input events. Except it infers "Handled" by you setting the Response, and I don't think we should make it more complicated by requiring you also set the Handled property.

I can't think of an Ing name for this event. The only thing I can think of is BasicAuthenticationRequired, but I'm OK with the current name.

Reply:
We've got examples of this already and don't have a better alternative.